WooCommerce/src/main/kotlin/com/woocommerce/android/util/PrintHtmlHelper.kt

```
@@ -0,0 +1,46 @@
+package com.woocommerce.android.util
+
+import android.app.Activity
+import android.content.Context
+import android.print.PrintAttributes
+import android.print.PrintManager
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import com.woocommerce.android.util.WooLog.T
+import javax.inject.Inject
+
+class PrintHtmlHelper @Inject constructor() {
+    // Hold an instance of the WebView object so it isn't garbage collected before the print job is created
+    private var webViewInstance: WebView? = null
+
+    fun printReceipt(activity: Activity, htmlReceipt: String, documentName: String) {
+        webViewInstance?.let {
+            WooLog.e(
+                T.UTILS,
+                "Initiating print job before the previous job has finished. " +
+                    "The previous job might fail since its WebView might get garbage collected."
+            )
+        }
+        val webView = WebView(activity)
+        webView.webViewClient = object : WebViewClient() {
+            override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest) = false
+
+            override fun onPageFinished(view: WebView, url: String) {
+                enqueuePrintJob(activity, view, documentName)
+                webViewInstance = null
+            }
+        }
+
+        webView.loadDataWithBaseURL(null, htmlReceipt, "text/HTML", "UTF-8", null)
+        webViewInstance = webView
+    }
+
+    private fun enqueuePrintJob(activity: Activity, webView: WebView, documentName: String) {
+        (activity.getSystemService(Context.PRINT_SERVICE) as? PrintManager)?.print(
```

is this a valid case that `PRINT_SERVICE` is not available in the system? If yes, shouldn't we show something to a user? Currently it will silently do nothing if I got it correct

---

I believe it's not a valid use case on Android 4+. Would you prefer if the app crashed in this scenario?

---

I would crash because we don't support android 4 anyway

---

Updated in 5e76dc6b1c2055294ecb0b705f0a436d5fc0dcec